packages/happy-dom/src/nodes/node/INode.ts

```diff
@@ -21,7 +22,10 @@ export default interface INode extends IEventTarget {
 	cloneNode(deep: boolean): INode;
 	appendChild(node: INode): INode;
 	removeChild(node: INode): void;
-	insertBefore(newNode: INode, referenceNode?: INode): INode;
+	insertBefore(newNode: INode, referenceNode?: INode | null): INode;
+	insertAdjacentElement(position: TInsertAdjacentPositions, node: INode): INode | null;
```

Comment:
insertAdjacentElement(), insertAdjacentHTML(), insertAdjacentText() is part of the IElement interface and should be implemented in the Element class.